src/reporters.ts

```diff
@@ -12,22 +12,41 @@ import { pipe } from "fp-ts/lib/function";
  *   ".a is not a number"
  *   ".c.b is not a non empty string"
  */
+
 const getContextPath = (context: Context): string => {
   if (context.length === 0) {
-    return "] (decoder info n/a)";
+    return " (decoder info n/a)";
   }
-  const keysPath = context.map(({ key }) => key).join(".");
+
+  /*
+    in order to get a more readable ouptut we want to avoid those keys that represent the index of a t.union or a t.intersection, 
+    to do this we want to render the key in the output just in 2 cases: 
+      1. the key can be parsed as a valid integer and the element right before was an ArrayType
+      2. the key is not parsable as a valid integer but it is also a non empty string
+  */
+  const keysPath = context
+    .filter(
+      // eslint-disable-next-line
+      (c: any, i: number) =>
+        (!isNaN(parseInt(c.key, 10)) &&
+          context[i - 1] &&
+          // eslint-disable-next-line
+          (context[i - 1].type as any)._tag === "ArrayType") ||
+        (isNaN(parseInt(c.key, 10)) && c.key !== "")
```

Comment:
parseInt use the decimal base as default, maybe we can avoid to specify it

Reply:
You are right, i have just added it cause of linter error